soufi/tests/finders/test_yum_finder.py

```diff
@@ -100,6 +100,22 @@ def test_find(self):
         self.assertIsInstance(disc_source, yum.YumDiscoveredSource)
         self.assertEqual([url], disc_source.urls)
 
+    def test_generate_repos_with_plain_list(self):
+        fallback = mock.MagicMock()
+        finder = self.make_finder()
+        repos = [self.factory.make_string() for _ in range(0, 3)]
+        result = finder.generate_repos(repos, fallback)
+        self.assertThat(result, SameMembers(repos))
+        fallback.assert_not_called()
+
+    def test_generate_repos_with_fallback(self):
+        fallback = mock.MagicMock()
+        repos_fallback = [self.factory.make_string() for _ in range(0, 3)]
+        fallback.return_value = repos_fallback
+        finder = self.make_finder()
+        result = finder.generate_repos(None, fallback)
+        self.assertThat(list(result), SameMembers(repos_fallback))
```

Comment:
Why the coercion to `list` here?

Reply:
The test was failing without it and I couldn't work out why, but I *think* SameMembers has trouble with empty generators.

Reply:
Weird.